moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -205,30 +233,32 @@ bool Servo::validateParams(const servo::Params& servo_params) const
     params_valid = false;
   }
 
+  if (!servo_params.active_subgroup.empty() && servo_params.active_subgroup != servo_params.move_group_name &&
+      !joint_model_group->isSubgroup(servo_params.active_subgroup))
+  {
+    RCLCPP_ERROR(LOGGER,
+                 "The value '%s' Parameter 'active_subgroup' does not name a valid subgroup of joint group '%s'.",
+                 servo_params.active_subgroup.c_str(), servo_params.move_group_name.c_str());
+    params_valid = false;
+  }
+
   return params_valid;
 }
 
 bool Servo::updateParams()
 {
   bool params_updated = false;
-
   if (servo_param_listener_->is_old(servo_params_))
   {
-    auto params = servo_param_listener_->get_params();
+    const auto params = servo_param_listener_->get_params();
 
-    const bool params_valid = validateParams(params);
-    if (params_valid)
+    if (validateParams(params))
```

Comment:
It's a bit of an issue here that `validateParams()` is expensive. It doesn't really belong in a control loop. My suggestion here would be:

1. Have an `initialParamValidation()` function that gets called once and does a full validation.
2. Have a `runtimeParamValidation()` function which only checks the things that can change, from here:  https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/config/servo_parameters.yaml

Preferably no string comparison or other expensive operations in `runtimeParamValidation()`.

Reply:
If we need to trim down the list of parameters that can change so they can be validated in realtime, that's fine with me. Open to other ideas.

Reply:
Also, simply not validating the parameters at all once the loop has started is a pretty good option.

Reply:
`validateParams()`  is only called within `updateParams()` when a parameter has actually changed.
https://github.com/ros-planning/moveit2/blob/739afb649f5e248cb7d049bb35348a57fe5e5287/moveit_ros/moveit_servo/src/servo.cpp#L215
That should be fine right?

Reply:
> * Have an `initialParamValidation()` function that gets called once and does a full validation.
> * Have a `runtimeParamValidation()` function which only checks the things that can change, from here:

But I like this idea better

Reply:
Made a new issue for this.

https://github.com/ros-planning/moveit2/issues/2412